repositories/repositories.go

```diff
@@ -68,9 +67,6 @@ func NewRepositories(
 		UserRepository: &UserRepositoryPostgresql{
 			transactionFactory: transactionFactory,
 		},
-		ApiKeyRepository: &ApiKeyRepositoryImpl{
-			transactionFactory: transactionFactory,
-		},
```

Comment:
I removed the useless `ApiKeyRepositoryImpl`, taking inspiration from :
- what Chris present in the `postgress` directory
- what latest repositories (like inbox one) done

In the end:
- move `ApiKeyRepository` interface def were it's needed
- implement directly on `MarbleDbRepository` all those interfaces

IMO, final questions on the repository topic are more or less :
- should we still be using squirel + transaction functions or plain SQL when possible ? (or in between)
- should we move the transaction from param to interface implementing repositories (more or less like what is done by Chris) moving out the sugar function 
```go
pgTx := repo.transactionFactory.adaptMarbleDatabaseTransaction(ctx, tx)
```

Reply:
Great ! 
My take on the two open questions:
- plain sql vs squirrel, I don't have a strong opinion either way - I can live with both, I just think that squirrel serves an important niche in the "mid complexity" queries with templated parameters. Especially for update queries, "where" conditions
- transaction passing: Here I have to say I disagree with Chris and I really like having one method to call, nevermind whether I want to execute it in a transaction or not. Certainly we can make `adaptMarbleDatabaseTransaction` a bit nicer to use (but then again there are probably limits to what we can do) but I prefer this to the other way

Reply:
Ok, I missred Chris code. What I though was more or less :
- same abstraction as current code (meaning a single endpoint for each repository method, with or without transactions)
- transaction handling done outside the repository method : meaning no more `adaptMarbleDatabaseTransaction` done inside each function on the injected `tx` parameter. 

What I had in mind is more or less described by:
```go
// In repository
func (repo *MarbleDbRepository) UpdateUser(ctx context.Context, userId string) error {
    pgTx := repo.transactionFactory.adaptMarbleDatabaseTransaction(ctx)
    
    // do internal logic here,
}

// in usecase
func (usecase *UserUseCase) UpdateUser(ctx context.Context, updateUser models.UpdateUser) (models.User, error) {
	tx, err := usecase.transactionFactory.Begin(ctx, models.DATABASE_MARBLE_SCHEMA)
	defer tx.Rollback(ctx) // should be wrapped inside our Begin implementation but you have the idea

        userRepository, err :=  usecase.getUserRepository(tx) // but by writting this, it feels not necessarly usefull compared to `tx` param injection
	if err != nil {
		return models.User{}, err
	}
	user, err := userRepository.UserByID(ctx, tx, updateUser.UserId)
	if err != nil {
		return models.User{}, err
	}
	if err := usecase.enforceUserSecurity.UpdateUser(user); err != nil {
		return models.User{}, err
	}
	if err := userRepository.UpdateUser(ctx, tx, updateUser); err != nil {
		return models.User{}, err
	}
	user, err = userRepository.UserByID(ctx, tx, updateUser.UserId)
	if err != nil {
		return models.User{}, err
	}

	tx.Commit(ctx)
	analytics.TrackEvent(ctx, models.AnalyticsUserUpdated, map[string]interface{}{"user_id": updatedUser.UserId})

	return updatedUser, nil
}
```

Reply:
FYI, digging inside `pgx` issues to understand how transaction actually work (= can we safelly defer `Rollback`) I found [this issue](https://github.com/jackc/pgx/issues/1667) that may be interesting for us.

As we wrap every call (even simple `SELECT` ones) inside a transaction, it may add a lot of roundtrips to our decision computation (when reading the DB). It's difficult to pronounce w\ measuring but removing tx on simpler call could result in a faster execution (at least, until the lazy thing is implemented inside pgx)

Reply:
Well in fact we're already doing the transaction handling (AKA starting it & committing it) in the usecase, the repo just has an adapter function. Which we can perhaps simplify a bit, but I'm not sure there is that much "fat" to skim on it.